File: OutsurancePython/pages/loginpage.py
from playwright.sync_api import Page

class LoginPage:
    def __init__(self, page: Page) -> None:
        self.page = page
        self.login_button = page.locator(".header-links a[href]").get_by_text("Log in")
        self.email_input = page.locator("input.email")
        self.password_input = page.locator("input.password")
        self.signin_button = page.locator("input.login-button")

    def load(self) -> None:
        self.page.goto('https://demowebshop.tricentis.com/')

    def loginaction(self, username: str, password: str) -> None:
        self.login_button.click()
        self.email_input.fill(username)
        self.password_input.fill(password)
        self.signin_button.click()

File: playwright-xunit/PrimeService.Tests/Pages/LoginPage.cs
using Microsoft.Playwright;
using System.Threading.Tasks;

namespace Outsource.Tests.Pages
{
    public class LoginPage
    {
        private IPage _page;
        private readonly ILocator _loginlink;
        private readonly ILocator _email;
        private readonly ILocator _password;
        private readonly ILocator _loginbutton;
        public LoginPage(IPage page) 
        {
            _page = page;
            _loginlink = _page.Locator(".header-links a[href]").GetByText("Log in");
            _email = _page.Locator("input.email");
            _password = _page.Locator("input.password");
            _loginbutton = _page.Locator("input.login-button");
        }

        public async Task ClickLogin() => await _loginlink.ClickAsync();

        public async Task LoginAction(string email, string password)
        {
            await _email.FillAsync(email);
            await _password.FillAsync(password);
        }

        public async Task ClickSignIn() => await _loginbutton.ClickAsync();
    }
}
